test/translator/common_test.cc

```diff
@@ -78,4 +78,65 @@ TEST(Common, ShouldCorrectlyWriteValueToSpan) {
   EXPECT_EQ(0b11000100, buffer[0]);
 }
 
+TEST(Common, ShouldBuildAllocationWithSuccessStatus) {
+  translator::Allocation allocation;
+  auto alloc_callback = [](uint16_t count) -> uint64_t {
+    if (count == 1) {
+      return 1337;
+    }
+    if (count == 3) {
+      return 7331;
+    }
+    return 0;
+  };
+  void (*dealloc_callback)(uint64_t, uint16_t) = nullptr;
+  translator::SetAllocPageCallbacks(alloc_callback, dealloc_callback);
```

Comment:
Is there any mechanism preventing race conditions when setting this in multiple tests?

Reply:
There currently isn't a mechanism for this, but I could definitely see value in that to prevent flakiness. Do you have any suggestions?

Reply:
On review it looks like gtest doesn't have any built-in parallelism, and environments that apply parallelism to it do so via multiple processes. so this probably isn't an issue after all.